File: Bug_Hunt_Practice/Question 4/solution.cpp
#include <iostream>
#include <iomanip> // For setting precision
using namespace std;

double calculateAverage(int arr[], int size) {
    int sum = 0;
    for (int i = 0; i < size; i++) {
        sum += arr[i];
    }
    double average = sum / size;
    return average;
}

int main() {
    int grades[] = {5, 10, 2};
    int size = 3;
    
    double avg = calculateAverage(grades, size);
    cout << "Expected Output: 5.666..." << endl;
    cout << "Actual Output:   " << setprecision(5) << avg << endl;
    return 0;
}

File: Bug_Hunt_Practice/Question 4/solution.py
def calculate_average(arr):
    total_sum = 0
    for num in arr:
        total_sum += num
    
    if not arr:
        return 0
        
    average = total_sum // len(arr)
    return average

# --- Main execution ---
grades = [5, 10, 2]
avg = calculate_average(grades)

print(f"Expected Output: 5.666...")
print(f"Actual Output:   {avg}")